### PR TITLE
Metadata Endpoints

### PR DESCRIPTION
**Pull request by vforgione**
- Created CBVs for Metadata Models
- Updated serializers, deserializers, fields to accompany
- Updated meta base model methods to assist in generic CBV logic

### CODE REVIEW

File: plenario_core/models.py [hunk 1]
```diff
@@ -60,10 +60,40 @@ def save(self, *args, **kwargs) -> None:
...
+        raise NotImplementedError
+
     @classmethod
-    def get_serializer_class(cls) -> Schema:
+    def get_deserializer_class(cls) -> Type[Schema]:  # NOQA
         raise NotImplementedError
 
+    @classmethod
+    def deserialize(
+            cls, payload: dict, instance: 'MetaBase'=None) -> 'MetaBase':
```
Comment (HeyZoos) [inline plenario_core/models.py]: Typo, it looks like this should be `MetaBase`

File: plenario_core/utils/field_parsing.py [hunk 2]
```diff
@@ -74,216 +69,96 @@
...
+
+    for src_fname, ftype in user_definitions.items():
+        # get the basics for creating the django field
+        try:
+            django_type = _ftype_to_django.get(ftype)
+        except TypeError:  # unhashable types
+            django_type = None
+        if not django_type:
+            errors[src_fname] = f'{ftype} is not a valid type.'
+            continue
```
Comment (HeyZoos) [inline plenario_core/utils/field_parsing.py]: The `django_type = None` and `if not` seem unnecessary here.

File: plenario_core/models.py (hunk 1 above)
Comment (vforgione) [reply plenario_core/models.py]: What?

Comment (HeyZoos) [reply plenario_core/models.py]: The type hint and return type of the method signature is a string `'MetaBase'`

Comment (HeyZoos) [reply plenario_core/models.py]: Instead of the class `MetaBase`

File: plenario_core/utils/field_parsing.py (hunk 2 above)
Comment (vforgione) [reply plenario_core/utils/field_parsing.py]: Catching the `TypeError` is important - if someone passes a number or a dictionary or some other unhashable type as a value, the call to `_ftype_to_django.get(ftype)` would blow up. If a user passes a string, say `barf` as the field type, it won't be found in the `_ftype_to_django` dictionary.

These are both important guards.

File: plenario_core/models.py (hunk 1 above)
Comment (HeyZoos) [reply plenario_core/models.py]: Ah, my bad.

Comment (vforgione) [reply plenario_core/models.py]: https://www.python.org/dev/peps/pep-0484/#runtime-or-type-checking

**Pull request merged**